### ISSUE TITLE
Zoom: Artifacts in scaling #305 & #18 (from the website project)

### ISSUE THREAD

**Issue by itsmedavep** [surge-synthesizer/surge#400]
**Describe the bug**
Per @baconpaul the bitmap scale and the underlying fill scale may be off at zoom levels other than 100%. 

This issue references conversations going on in #305 and [#18 from the website project ]((https://github.com/surge-synthesizer/surge-synthesizer.github.io/issues/18))

**To Reproduce**
1. Open Surge
2. Click menu button
3. Pick zoom level of 125%
4. See this:
<img width="650" alt="screen shot 2019-01-27 at 12 34 49 pm" src="https://user-images.githubusercontent.com/11078351/51805182-f58d5b80-222f-11e9-9237-f2920989779a.png">

**Expected behavior**
The grey box should be centered within the black rect. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS 10.14
 - Host: Logic
 - 1.6

**Additional context**
Im not sure this is related but I see this as well at the same zoom level:
<img width="73" alt="screen shot 2019-01-27 at 12 37 51 pm" src="https://user-images.githubusercontent.com/11078351/51805232-89f7be00-2230-11e9-8702-dfc78a08b3d5.png">

The aliasing in the very first screenshot is concerning to me as well. At this zoom level are we still using the 1x assets or are we grabbing 1.5x and downsizing? Im trying to work out why those vectors are aliasing they way they are in that image. Id like to fix if possible on my end. I first need to know what asset were grabbing at the zoom level to start a visual diff on the rendered file vs the one that is being drawn in the GUI.

**Comment by baconpaul** [comment]
At zoom leave of 1.25x on a Retina display we will grab the 3x assets and downsize to 250; and on a non Retina display we will grab the 1.5 and downsize to 1.25.

**Comment by itsmedavep** [comment]
Cheers. Ill crack those open and take a look at what is going on with the aliasing in those rendered assets. 

Something about that top screen shot seems off to me. I swear I was enforcing snapping to pixel bounds with the vectors in the design tool. 

I know we could be drawing at sub pixels  and that could introduce the aliasing but dang does it seems like a lot in that screen grab.

**Comment by baconpaul** [comment]
Yeah don’t kill yourself until I get a chance to read the code. I bet we are drawing to a size -1 in float space and should draw to a size -0.5 or some such

Vst3sdk pointer and bitmap arity come first tho

**Comment by itsmedavep** [comment]
I was thinking it could be something like that. macOS requires a similar approach as well. 

Drawing controls with Cocoa in macOS requires a half pixel transform and then a backstop of rounding to nearest pixel to keep everything crisp. Or at least it did the last time I was doing much of that type of work. 

TL;DR
Ill hold off on the spelunking till its warranted.

**Comment by esaruoho** [comment]
![fullscreen_30_01_2019__15_44](https://user-images.githubusercontent.com/4966687/51985382-62119000-24a6-11e9-96ad-22a54babc18e.jpg)

posting some red arrows here to highlight this weird line that appears below the sliders :)

**Comment by baconpaul** [comment]
OK this is a deeply annoying-sh problem

The way this works is it sub-samples bits of image 136 into image 137 trimming and moving it around. Since 137 has white background if that copy in is off by just a little when scaled, it blows up.

So the most correct thing to do - I think - is to bump the black borders in about 1/4 pixel in the 300% image.

Alternately this diff

```
paul:~/dev/music/surge$ git diff
diff --git a/src/common/gui/CEffectSettings.cpp b/src/common/gui/CEffectSettings.cpp
index bcd319f..740c2f3 100644
--- a/src/common/gui/CEffectSettings.cpp
+++ b/src/common/gui/CEffectSettings.cpp
@@ -67,7 +67,7 @@ void CEffectSettings::draw(CDrawContext* dc)
       for (int i = 0; i < 8; i++)
       {
          CRect r(0, 0, 17, 9);
-         r.offset(size.left, size.top);
+         r.offset(size.left+0.1, size.top);
          r.offset(blocks[i][0], blocks[i][1]);
          int stype = 0;
          if (disabled & (1 << i))
```

can get us here

<img width="902" alt="screen shot 2019-02-01 at 8 05 23 pm" src="https://user-images.githubusercontent.com/13387561/52157389-b65e7080-265c-11e9-8759-b0f3a24f9b27.png">

at a 300% zoom on 300% zoom

Oh and @itsmedavep it seems that arrow going into the plus is still needing vectorizification!

**Comment by itsmedavep** [comment]
Ohhhhhh. Thats what the comment meant!!!! I changed all of the arrows in the graphic out and didnt realize that is mean to be another arrow there. 

Just so Im sure were talking about the same thing:

![screen shot 2019-02-02 at 11 38 53 am](https://user-images.githubusercontent.com/11078351/52167377-b6ec1b00-26df-11e9-9906-4a3cab47250b.png)

Its supposed to be an arrow POINTING to the "+". Easy to update.

**Comment by baconpaul** [comment]
Yeah or something
Also the line going back to the 0 before seems to be 2 lines. 
Basically that whole area is you know. Sort of wonky

**Comment by itsmedavep** [comment]
> Yeah or something
> Also the line going back to the 0 before seems to be 2 lines.
> Basically that whole area is you know. Sort of wonky

Got it! That make sooo much sense. It was a 💡 moment there for me. Cool. Ill fix up now!

**Comment by mkruselj** [comment]
Possibly related, too - these lines show up at non-100% zoom level over here (W10):

![image](https://user-images.githubusercontent.com/2393720/52169372-5a1c4f00-2737-11e9-8b4d-a51e82bf1523.png)

**Comment by itsmedavep** [comment]
Thanks, for further context:

![screen shot 2019-02-02 at 3 11 22 pm](https://user-images.githubusercontent.com/11078351/52169380-db0a2500-26fc-11e9-89c7-1cc7058ee080.png)

**Comment by baconpaul** [comment]
<img width="1270" alt="screen shot 2019-02-24 at 9 36 18 am" src="https://user-images.githubusercontent.com/13387561/53300600-a844df80-3817-11e9-818d-b9c14343ddb9.png">

With the SVG rendering this is mostly fixed, btw. So Lemme couple this to #647
